### ISSUE TITLE
{{account_id}} works different with and without lambda

### ISSUE THREAD

**Issue by mkadali99** [capitalone/cloud-custodian#3144]
The {{account_id}} to reflect the account's id in the email is behaving different when I choose to trigger lambda and when I run it manually. Any clues? I have tried {{account_id}} and {{account}} as well.

 mode:
 type: periodic
 schedule: rate(15 minutes)

(The lambda being included)

 subject: "EBS Lifecycle Tag removed -AWS Acc {{account_id}} in {{region}}"

**Comment by kapilt** [comment]
different how? what version of custodian and mailer? could you provide a complete policy example?

**Comment by mkadali99** [comment]
Yeah. When I run the policy manually the email notification reflects the account id 
This is how the subject of my email looks like
"EBS Lifecycle Tag removed -AWS Acc xxxxxxxxx in us-east-1

But when I include the mode in the policy and let the lambda take care of it, my subject looks like this 
"EBS Lifecycle Tag removed -AWS Acc  in us-east-1

**Comment by mkadali99** [comment]
Here is the example policy
policies:
- name: EBS-Lifecycle-Autotag
  resource: ebs
  mode:
   type: periodic
   schedule: rate(24 hours)
   role: arn:aws:iam::xxxxxxxxxxxx:role/example_role
  description:
    Ensure all volumes have the ebs lifecycle tag
  filters:
    - not:
      - "tag:ebs lifecycle": "daily_14_version"

  actions:
    - type: notify
      template: default.html
      subject: "EBS-Lifecycle-Autotag-new EBS Lifecycle Tag Missing -AWS Acc:{{ account_id }}"
      violation_desc: "Volume does not have the 'ebs lifecycle' tag"
      action_desc: "ebs lifecycle tag added"
      to:
        - resource-owner
        - someone@domain.com
        
      transport:
          type: sqs
          queue: https://sqs.us-east-1.amazonaws.com/xxxxxxxxxxx/queue_name
          region: us-east-1
    - type: mark
      key: ebs lifecycle
      value: "daily_14_version"

**Comment by mkadali99** [comment]
Here is the example policy
policies:
- name: EBS-Lifecycle-Autotag
  resource: ebs
 mode:
   type: periodic
   schedule: rate(24 hours)
   role: arn:aws:iam::xxxxxxxxxxxx:role/role_name
  description:
    Ensure all volumes have the ebs lifecycle tag
  filters:
    - not:
      - "tag:ebs lifecycle": "daily_14_version"
    - or:
      - "tag:cpm backup": "absent"
      - "tag:cpm backup": "empty"
      - "tag:cpm backup": "False"
      - "tag:cpm backup": "no-backup"
  actions:
    - type: notify
      template: default.html
      subject: "EBS-Lifecycle-Autotag-new EBS Lifecycle Tag Missing -AWS Acc:{{ account_id }}"
      violation_desc: "Volume does not have the 'ebs lifecycle' tag"
      action_desc: "ebs lifecycle tag added"
      to:
        - resource-owner
        - someone@domain.com
       
      transport:
          type: sqs
          queue: https://sqs.us-east-1.amazonaws.com/xxxxxxxxxxxx/queue_name 
          region: us-east-1
    - type: mark
      key: ebs lifecycle
      value: "daily_14_version"

**Comment by kapilt** [comment]
i suspect this might be we're evaluating account_id when we're provisioning instead of when we're executing the lambda.

**Comment by gtrindade** [comment]
This seems to still be an issue, `account_id` gives me the id correctly, but `account` is blank. Is there a workaround or plans to fix this?

**Comment by kapilt** [comment]
definitely game for contributions to resolve if your interested.
